toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/utils/ConfigTesterHelper.java

```diff
@@ -0,0 +1,221 @@
+			if (runTryValue) {
+				NotificationManager.showMessage(url + " has spaces in its name.").showNotification();
+			}
+		}
+
+		if (getStatus(url,proxyAdress, proxyPort, proxyUserName, proxyUserPswd)) {
+			isKo = false;
+		} else {
+			if (runTryValue) {
+				NotificationManager.showMessage(url + " does not answer.").showNotification();
+			}
+		}
+		return !isKo;
+	}
+
+	public static boolean getStatus(String url, String proxyAdress, String proxyPort, String proxyUserName,
+			String proxyUserPswd) throws IOException {
+
+	        boolean result = false;
+	        Proxy proxy = null;
+	        if(proxyAdress != null && proxyPort != null) {
+				InetSocketAddress proxyInet = new InetSocketAddress(proxyAdress,Integer.parseInt(proxyPort));
+				proxy = new Proxy(Proxy.Type.HTTP, proxyInet);
+	        }
+	        
+	        Authenticator.setDefault (new Authenticator() {
+	            protected PasswordAuthentication getPasswordAuthentication() {
+	                return new PasswordAuthentication (proxyUserName, proxyUserPswd.toCharArray());
+	            }
+	        });
+
+			try {
+				HttpURLConnection connection;
+				if(url.startsWith("https")){
+					connection = pingHttpsUrl(url, proxy);
+				} else {
+					connection = pingHttpUrl(url, proxy);
+				}
+				int code = connection.getResponseCode();
+				if(code == 301){
+					result = getStatus(connection.getHeaderField("Location"), proxyAdress, proxyPort, proxyUserName, proxyUserPswd);
+				} else if (code == 200) {
+					result = true;
+				}
+			} catch(SSLHandshakeException sslException){
+	            try {
+					HttpsURLConnection connection = pingHttpsUrl(url, proxy);
+					int code = connection.getResponseCode();
+					if(code == 301){
+						result = getStatus(connection.getHeaderField("Location"), proxyAdress, proxyPort, proxyUserName, proxyUserPswd);
+					} else if (code == 200) {
+						result = true;
+					}
+	            } catch (Exception e) {
+	                result = false;
+	            }
+	        } catch (SocketTimeoutException e) {
+				return false;
+			}
+			catch (Exception e) {
+	            result = false;
+	        }
+	        return result;
+	}
+	
+	public static HttpURLConnection pingHttpUrl(String url, Proxy proxy) throws IOException {
+		URL siteURL = new URL(url);
+		HttpURLConnection connection;
+		if(url.contains("localhost") || proxy == null){
+			connection = (HttpURLConnection) siteURL.openConnection();
+		} else {
+			connection = (HttpURLConnection) siteURL.openConnection(proxy);		
+		}
+		connection.setRequestMethod("GET");
+		connection.setConnectTimeout(timeout);
+		connection.connect();
+		return connection;
+	}
+
+	public static HttpsURLConnection pingHttpsUrl(String url, Proxy proxy) throws IOException, NoSuchAlgorithmException, KeyManagementException {
+		TrustManager[] trustAllCerts = new TrustManager[]{
+				new X509TrustManager() {
+					public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+						return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Return an empty array instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)